libraries/rush-lib/src/logic/buildCache/AdditionalFilesHasher.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { Import } from '@rushstack/node-core-library';
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+
+import type { IOptions } from 'glob';
+
+const glob: typeof import('glob') = Import.lazy('glob', require);
+
+const globAsync = (pattern: string, options: IOptions = {}): Promise<string[]> => {
+  return new Promise((resolve, reject) => {
+    glob(pattern, options, (err, matches) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve(matches);
+    });
+  });
```

Comment:
```suggestion
  return await LegacyAdapters.convertCallbackToPromise(glob, pattern, options)
```

`LegacyAdapters` comes from `node-core-library`